ChiModules/KEigenvalueSolver/initialize_precursors.cc

```diff
@@ -15,45 +15,48 @@ using namespace LinearBoltzmann;
 /**Computes the point wise delayed neutron precursor concentrations.*/
 void KEigenvalue::Solver::InitializePrecursors()
 {
-  typedef SpatialDiscretization_PWLD  PWLD;
-  auto pwl = std::static_pointer_cast<PWLD>(discretization);
+  if (options.use_precursors)
+  {
+    typedef SpatialDiscretization_PWLD  PWLD;
+    auto pwl = std::static_pointer_cast<PWLD>(discretization);
 
-  Nj_new_local.assign(Nj_new_local.size(), 0.0);
+    precursor_new_local.assign(precursor_new_local.size(), 0.0);
 
-  // ----- Loop over cells
-  for (auto& cell : grid->local_cells)
-  {
-    // ----- Cell information
-    const auto xs_id = matid_to_xs_map[cell.material_id];
-    auto& xs = material_xs[xs_id];
-    auto cell_fe_view = pwl->GetCellMappingFE(cell.local_id);
-    auto& transport_view = cell_transport_views[cell.local_id];
-
-    // ----- Loop over cell dofs
-    for (int i = 0; i < transport_view.NumNodes(); ++i)
+    //======================================== Loop over cells
+    for (auto& cell : grid->local_cells)
     {
-      size_t ir = transport_view.MapDOF(i, 0, 0);
-      size_t jr = pwl->MapDOFLocal(cell, i, Nj_unk_man, 0, 0);
-      double* Nj_newp  = &Nj_new_local[jr];
-      double* phi_newp = &phi_new_local[ir];
-
-      // ----- If a fissial material with precursors
-      if ((xs->is_fissile) and (xs->num_precursors > 0)) {
-        // ----- Loop over precursors
-        for (size_t j = 0; j < xs->num_precursors; ++j)
+      //==================== Cell information
+      const auto xs_id = matid_to_xs_map[cell.material_id];
+      auto& xs = material_xs[xs_id];
+      auto cell_fe_view = pwl->GetCellMappingFE(cell.local_id);
+      auto& transport_view = cell_transport_views[cell.local_id];
+
+      //============================== Loop over nodes
+      for (int i = 0; i < cell_fe_view->num_nodes; ++i)
+      {
+        size_t ir = transport_view.MapDOF(i, 0, 0);
+        size_t jr = pwl->MapDOFLocal(cell, i, precursor_uk_man, 0, 0);
+        double* Nj_newp = &precursor_new_local[jr];
```

Comment:
Just my opinion, but I think this is unnecessary and more convoluted than just adding an offset below. I also doubt that it's faster.

Reply:
Perhaps. This is a design that has been used since I began using ChiTech that is used in more places than this PR affects. I would suggest raising a separate issue for this.

Reply:
@Naktakala What are your thoughts on this?

Reply:
Yes, I've come to dislike my style here. Get rid of all the `somethingp` pointers. When we discussed this last week I've looked up the optimizers behavior in this regard. It pretty much optimizes out well.